core/interfaces/branch.py

```diff
@@ -502,18 +502,23 @@ def run(self, edit):
 
     def run_async(self):
         self.interface = ui.get_interface(self.view.id())
-        current_branch_name = self.get_current_branch_name()
 
+        branches = self.get_selected_branches()
+        branches_strings = self.create_branches_strs(branches)
+        self.merge(branches_strings) if not len(branches_strings) == 0 else None
+        util.view.refresh_gitsavvy(self.view)
+
+    def get_selected_branches(self):
+        current_branch_name = self.get_current_branch_name()
         branches = set()
         for sel in self.view.sel():
-            for name in self.get_selected_branches(sel):
+            for name in self.get_selected_branch(sel):
                 if name and not name == current_branch_name:
                     branches.add(name)
 
-        self.merge(branches) if not len(branches) == 0 else None
-        util.view.refresh_gitsavvy(self.view)
+        return branches
 
-    def get_selected_branches(self, sel):
+    def get_selected_branch(self, sel):
```

Comment:
Personally, I'd consolidate `get_selected_branch` and `get_selected_branch_name` under `get_selected_branches`. In my opinion since they call each other and not used anywhere else, they can be combined into a single method that does only 1 thing (get all selected branch names, with remote optionally included).